hax/hax/motr/delivery.py

```diff
@@ -46,6 +46,24 @@ def __init__(self):
         self.waiting_clients: Dict[HaLinkMessagePromise, Condition] = {}
         self.lock = Lock()
 
+    def check_delivery(self,
```

Comment:
[suggestion]: s/check_delivery/is_delivered/

Reply:
My strong opinion is that the method name should start with `_` prefix (to mark that it is a private one, no one is expected to invoke it outside of this class; as a bonus no docstring is required for a private method 😁).

`_is_delivered()` looks better than `check_delivery` but it doesn't return a `bool` value (although `is` is usually a sign of boolean). So I would recommend something like `_ensure_delivered()`.

In fact the method does quite a lot of different things, so it is hard to name it properly. Let's just choose a name that would mislead the developers in a minimal way.